### DISCUSSION TITLE
Welcome to HRLDAS Community Discussion Forum!

### DISCUSSION THREAD

**Discussion by cenlinhe** [NCAR/hrldas#68] [Announcements]
## 👋 Welcome!
  We’re using this Discussion Forum as a place to connect with all members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  
  📢 **Announce to your community** that Discussions is available! Go ahead and send that tweet, post, or link it from the website to drive traffic here.

  🔗 If you use issue templates, **link any relevant issue templates** such as questions and community conversations to Discussions. Declutter your issues by driving community content to where they belong in Discussions. If you need help, here's a [link to the documentation](https://docs.github.com/en/github/building-a-strong-community/configuring-issue-templates-for-your-repository#configuring-the-template-chooser).

  ➡️ You can **convert issues to discussions** either individually or bulk by labels.

**Comment by MoreblessingMuzava** [comment]
**`create_forcing.exe` "No previous data" error for T2D with WRF Output (WRF-Hydro/HRLDAS)**

Hello everyone,
I am attempting to generate HRLDAS forcing files from WRF (ARW) model output, but I am consistently encountering a "No previous data" error for the `T2D` variable when running `create_forcing.exe`.
The program does not crash, but it stops processing further variables, indicating it's not generating complete forcing files. I have performed extensive troubleshooting, and I'm hoping someone with specific knowledge of `create_forcing.exe`'s internal workings or common incompatibilities might be able to offer guidance.

**My Goal:**
To successfully convert `wrfout_d01_YYYY-MM-DD_HH:MM:SS` files (containing 2m temperature as `T2D`) into HRLDAS forcing files using `create_forcing.exe`.

**My System & Software Setup:**
  * **Operating System:** Ubuntu (presumably the version of your VM)
  * **WRF Version:** WRF-ARW 4.5.2
  * **HRLDAS/WRF-Hydro Code:** Cloned from the official Git repository 
  * **Compiler:** `mpif90` 
  * **NetCDF Libraries:** Installed at `/opt/wrf_libs/netcdf/` (including `libnetcdf` and `libnetcdff`). These were explicitly linked during `create_forcing.exe` compilation.

**The Problem:**

When I run `./create_forcing.exe`, it initializes, reads `geo_em.d01.nc`, identifies the start date, but then prints "No previous data" for the `T2D` variable and does not proceed to process other variables.
**Exact `create_forcing.exe` Output:**
```
 Forcing type: 1                                                                                                                              
 2|100|1000|0|T2|K|"2m Temperature"|0|0|0|100
 ... (other variable listings)
 geo_em_flnm     = "/opt/wrf_libs/WPS/geo_em.d01.nc"
  flnm = /opt/wrf_libs/WPS/geo_em.d01.nc
 Done with subroutine read_geo_em_file
 Date = 2019-03-15_00  ihour =            0
 Field label= T2D
 No previous data.
```
**Troubleshooting Steps Taken (and their Results):**
I have thoroughly checked the following:
1.  **WRF Output File Variable Naming:**
      * Initially, the 2m temperature variable in `wrfout` files was `T2`.
      * I successfully renamed `T2` to `T2D` in *all* `wrfout` files using `ncrename -v T2,T2D`. `ncrename` reported success.
2.  **Verification of `T2D` Existence in `wrfout` Files:**
      * `ncdump -h wrfout_d01_2019-03-15_00:00:00 | grep "T2"` (and for other files) now clearly shows `float T2D(...)`.
      * The `:history` attribute in the `wrfout` files confirms the `ncrename` operation was successful.
3.  **Verification of `T2D` Data Content:**
      * `ncdump -v T2D -f c wrfout_d01_2019-03-15_00:00:00` confirms that the `T2D` variable contains valid numerical data (e.g., `T2D = 292.4706, 292.2442, ...`). The data is not missing or malformed according to `ncdump`.
4.  **`namelist.input` Configuration for `create_forcing.exe`:**
      * The `namelist.input` (located in `HRLDAS_forcing/`) correctly specifies the `DataDir` as `/opt/wrf_libs/WRF-4.5.2/run/` where the `wrfout` files reside.
      * `startdate` (`2019-03-15_00:00`) and `enddate` (`2019-03-17_12:00`) are set.
      * The `Tfile_template` matches the `wrfout` filename format: `'wrfout_d01_%Y-%M-%D_%H:00:00'`.
5.  **`create_forcing.exe` Compilation:**
      * I performed a `make clean` followed by `make` within the `HRLDAS_forcing/` directory.
      * Compilation was successful, explicitly linking against `/opt/wrf_libs/netcdf/lib` and using `/opt/wrf_libs/netcdf/include`.
      * A minor `tmpnam` warning from Jasper was observed but is non-fatal.
6.  **File Permissions:**
      * `ls -l /opt/wrf_libs/WRF-4.5.2/run/wrfout_d01_2019-03-15_00:00:00` shows `rw-rw-r--` permissions for the `ubuntu` user, confirming read access.

**My Question:**

Given that all checks on the input files and compilation seem correct, what could be causing `create_forcing.exe` to report "No previous data" for `T2D`? Are there any known subtle incompatibilities with specific NetCDF versions from WRF output, or internal assumptions within `create_forcing.exe` that I might be missing?

Any insights or suggestions would be greatly appreciated. Thank you for your time.
-----

**Reply by cenlinhe** [reply]
create_forcing does not work for wrfout  files. You have to modify the source code in create_forcing.F file to make it work. However, since you already have wrfout files, you could simply write a python/matlab/ncl script to extract the forcing variables from your wrfout  file and interpolate them into your HRLDAS grids (if your HRLDAS grid is different from WRF), and then generate the HRLDAS forcing files. You do not need to use create_forcing in this case.
